lib/cylc/batch_sys_manager.py

```diff
@@ -592,44 +600,59 @@ def _jobs_poll_status_files(self, job_log_root, job_log_dir):
 
     def _jobs_poll_batch_sys(self, job_log_root, batch_sys_name, my_ctx_list):
         """Helper 2 for self.jobs_poll(job_log_root, job_log_dirs)."""
-        batch_sys = self.get_inst(batch_sys_name)
-        all_job_ids = [ctx.batch_sys_job_id for ctx in my_ctx_list]
-        if hasattr(batch_sys, "get_poll_many_cmd"):
-            # Some poll commands may not be as simple
-            cmd = batch_sys.get_poll_many_cmd(all_job_ids)
-        else:  # if hasattr(batch_sys, "POLL_CMD"):
-            # Simple poll command that takes a list of job IDs
-            cmd = [batch_sys.POLL_CMD] + all_job_ids
-        try:
-            proc = Popen(cmd, stderr=PIPE, stdout=PIPE)
-        except OSError as exc:
-            # subprocess.Popen has a bad habit of not setting the
-            # filename of the executable when it raises an OSError.
-            if not exc.filename:
-                exc.filename = cmd[0]
-            sys.stderr.write(str(exc) + "\n")
-            return
-        proc.wait()
-        out, err = proc.communicate()
-        sys.stderr.write(err)
-        if hasattr(batch_sys, "filter_poll_many_output"):
-            # Allow custom filter
-            job_ids = batch_sys.filter_poll_many_output(out)
-        else:
-            # Just about all poll commands return a table, with column 1
-            # being the job ID. The logic here should be sufficient to
-            # ensure that any table header is ignored.
-            job_ids = []
-            for line in out.splitlines():
-                try:
-                    head = line.split(None, 1)[0]
-                except IndexError:
-                    continue
-                if head in all_job_ids:
-                    job_ids.append(head)
+        exp_job_ids = [ctx.batch_sys_job_id for ctx in my_ctx_list]
+        bad_job_ids = list(exp_job_ids)
+        exp_pids = []
+        bad_pids = []
+        items = [[self.get_inst(batch_sys_name), exp_job_ids, bad_job_ids]]
+        if getattr(items[0][0], "CAN_POLL_PROC_GROUP", False):
+            exp_pids = [ctx.pid for ctx in my_ctx_list if ctx.pid is not None]
+            bad_pids = bad_pids.extend(exp_pids)
+            items.append([self.get_inst("background"), exp_pids, bad_pids])
+        for batch_sys, exp_ids, bad_ids in items:
+            batch_sys = self.get_inst(batch_sys_name)
```

Comment:
This line needs to be deleted - it reverts `batch_sys` back to the original ("at").